### PR TITLE
Add minimal-raw and clean up the compose handler a bit

### PR DESCRIPTION
**Pull request by ondrejbudai**
Firstly, this adds the minimal-raw image type to the cloud API.

However, I wasn't sure how to test it - we surely shouldn't build all images in the composer's CI - that's what the images' CI is for. Since adding an image type to cloud API is about implementing a translation between cloud API and the image definitions API, I decided to clean up the translation code a bit to reduce its API surface and cover it with a smoke test. The test itself is pretty dumb, but that's just how tests for translation layers look like. :shrug:

### CODE REVIEW

File: internal/cloudapi/v2/server.go [hunk 1]
```diff
@@ -126,6 +126,9 @@ func (s *Server) enqueueCompose(distribution distro.Distro, bp blueprint.Bluepri
 	ir := irs[0]
 
 	ibp := blueprint.Convert(bp)
+	// shortcut
+	arch := ir.imageType.Arch()
```
Comment (achilleas-k) [inline internal/cloudapi/v2/server.go]: The comment here doesn't really help much.  I think it's clear what's going on from the code.

**Review by achilleas-k — APPROVED** [review]
I love the API simplifications here.  More of this!! <3

LGTM

**Comment by thozza** [comment]
The nightly pipeline failure is a known issue tracked in https://issues.redhat.com/browse/RHEL-24346

**Comment by thozza** [comment]
The failure in the edge test case is a known issue, as confirmed by https://github.com/osbuild/osbuild-composer/pull/3828#issuecomment-1952384801.

The failure in the nightly pipeline is a known issue, tracked in https://issues.redhat.com/browse/RHEL-24346

All the required tests passed. Merging manually (@nullr0ute asked to get this in ASAP), since there is an unresolved discussion, which I want to keep open.

**Pull request merged**